apps/main/src/lend/entities/market-onchain-rate.ts

```diff
@@ -0,0 +1,32 @@
+import { lendingJsValidationSuite } from '@/lend/entities/validation/lending-js-validation'
+import { ChainId } from '@/lend/types/lend.types'
+import type { ChainParams, ChainQuery } from '@ui-kit/lib/model/query'
+import { queryFactory } from '@ui-kit/lib/model/query'
+import { useApiStore } from '@ui-kit/shared/useApiStore'
+
+type Params = ChainParams<ChainId> & {
+  marketId: string
+}
+
+const _fetchOnChainMarketRate = async ({ chainId, marketId }: ChainQuery<ChainId> & { marketId: string }) => {
+  const api = useApiStore.getState().lending
+  const market = api?.getOneWayMarket(marketId)
+
+  if (!market) return null
```

Comment:
I don't think this is reachable, api is validated in the suite and it will throw an error if the marketId is unknown